### PR TITLE
Make VonMises more flexible

### PR DESCRIPTION
**Pull request by cscherrer**
# Parameterized Trait Enhancement & VonMises Scaling Support

## Summary
This PR introduces parameter structs for distributions and enhances the `Parameterized` trait with a default `map_params` implementation. It also adds scaling support for the VonMises distribution.

### 1. Parameter Structs
- Added parameter structs for all distributions (e.g., `BernoulliParameters`, `GaussianParameters`)
- Standardized parameter representation for better type safety and clarity
- Updated all `emit_params` and `from_params` implementations to use these structs

### 2. Enhanced `Parameterized` Trait
- Added a default `map_params` implementation that enables mapping functions over parameters
- This provides a consistent way to transform distribution parameters

### 3. VonMises Distribution Improvements
- Added `Scalable` trait implementation for VonMises distribution
- Added `map_params` implementation for efficient parameter updates
- Added `from_parts_unchecked` for more efficient construction

### 4. Data Module Enhancements
- Added extensive tests for `DataOrSuffStat` and categorical datum
- Fixed Scaled distribution representation by moving to eager computation of log Jacobian

## Testing
Added comprehensive test coverage for all new functionality, including:
- Edge cases for categorical data
- Boolean conversions
- Parameter mapping
- VonMises scaling operations
- Additional data module validation

### CODE REVIEW

File: src/traits.rs [hunk 1]
```diff
@@ -8,6 +8,19 @@ pub trait Parameterized {
     fn emit_params(&self) -> Self::Parameters;
 
     fn from_params(params: Self::Parameters) -> Self;
+
+    // TODO: If Self: Sized ok here? Should this be a requirement for the whole trait?
```
Comment (schmidmt) [inline src/traits.rs:12]: I think in every case the parameters should be sized.

**Review by schmidmt — APPROVED** [review]
This all seems reasonable. I like moving to named parameters too.

File: src/traits.rs (hunk 1 above)
Comment (cscherrer) [reply src/traits.rs]: Ok I changed it to 
```rust

pub trait Parameterized: Sized {
    type Parameters;

    fn emit_params(&self) -> Self::Parameters;

    fn from_params(params: Self::Parameters) -> Self;

    fn map_params(
        &self,
        f: impl Fn(Self::Parameters) -> Self::Parameters,
    ) -> Self {
        let params = self.emit_params();
        let new_params = f(params);
        Self::from_params(new_params)
    }
}
```
